plugins/modules/vmware_cluster_dpm.py

```diff
@@ -0,0 +1,188 @@
+        if self.cluster is None:
+            self.module.fail_json(msg="Cluster %s does not exist." % self.cluster_name)
+
+    def check_dpm_config_diff(self):
+        """
+        Check DRS configuration diff
+        Returns: True if there is diff, else False
+
+        """
+        dpm_config = self.cluster.configurationEx.dpmConfigInfo
+        change_message = None
+        changes = False
+
+        if dpm_config.enabled != self.enable_dpm:
+            change_message = 'DPM enabled status changes'
+            changes = True
+            return changes, change_message
+        elif self.enable_dpm:
+            if dpm_config.hostPowerActionRate != self.host_power_action_rate or dpm_config.defaultDpmBehavior != self.default_dpm_behaviour:
+                change_message = 'DPM Host Power Action Rate and/or default DPM behaviour change.'
+                changes = True
+                return changes, change_message
+
+        return changes, change_message
+
+    def configure_dpm(self):
+        """
+        Manage DRS configuration
+
+        """
+        changed, result = self.check_dpm_config_diff()
+
+        if changed:
+            if not self.module.check_mode:
+                cluster_config_spec = vim.cluster.ConfigSpecEx()
+                cluster_config_spec.dpmConfig = vim.cluster.DpmConfigInfo()
+                cluster_config_spec.dpmConfig.enabled = self.enable_dpm
+                cluster_config_spec.dpmConfig.defaultDpmBehavior = self.default_dpm_behaviour
+                cluster_config_spec.dpmConfig.hostPowerActionRate = self.host_power_action_rate
+
+                try:
+                    task = self.cluster.ReconfigureComputeResource_Task(cluster_config_spec, True)
+                    changed = wait_for_task(task)[0]
+                except vmodl.RuntimeFault as runtime_fault:
+                    self.module.fail_json(msg=to_native(runtime_fault.msg))
+                except vmodl.MethodFault as method_fault:
+                    self.module.fail_json(msg=to_native(method_fault.msg))
+                except TaskError as task_e:
+                    self.module.fail_json(msg=to_native(task_e))
+                except Exception as generic_exc:
+                    self.module.fail_json(msg="Failed to update cluster"
+                                              " due to generic exception %s" % to_native(generic_exc))
+            else:
+                changed = True
+
+        self.module.exit_json(changed=changed, result=result)
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(
+        cluster_name=dict(type='str', required=True),
+        datacenter=dict(type='str', required=True, aliases=['datacenter_name']),
+        enable_dpm=dict(type='bool', default=False),
+        default_dpm_behaviour=dict(type='str', choices=['automated', 'manual'], default='automated'),
+        host_power_action_rate=dict(type='int', choices=range(1, 6), default=3)
```

Comment:
```suggestion
        host_power_action_rate=dict(type='int', choices=[1, 2, 3, 4, 5], default=3)
```

Looks like ansible doesn't like `range(1, 6)` here.